public/enhanced-table-vis.js

```diff
@@ -1,32 +1,46 @@
-import 'plugins/enhanced-table/enhanced-table-vis.less';
-import 'plugins/enhanced-table/enhanced-table-vis-controller';
-import 'plugins/enhanced-table/enhanced-table-vis-params';
+import './enhanced-table-vis.less';
+import './enhanced-table-vis-controller';
+import './enhanced-table-vis-params';
+import 'ui/agg_table';
+import 'ui/agg_table/agg_table_group';
+import tableVisTemplate from './enhanced-table-vis.html';
+import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
+import image from './images/icon-table.svg';
 
 import { VisVisTypeProvider } from 'ui/vis/vis_type';
 import { TemplateVisTypeProvider } from 'ui/template_vis_type/template_vis_type';
 import { VisSchemasProvider } from 'ui/vis/schemas';
+// we need to load the css ourselves
 
-import visTemplate from 'plugins/enhanced-table/enhanced-table-vis.html';
-import paramsTemplate from 'plugins/enhanced-table/enhanced-table-vis-params.html';
+// we also need to load the controller and used by the template
 
-import image from './images/icon-table.svg';
+// our params are a bit complex so we will manage them with a directive
 
-import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
+// require the directives that we use as well
 
+// register the provider with the visTypes registry
 VisTypesRegistryProvider.register(EnhancedTableVisProvider);
 
+// define the TableVisType
 function EnhancedTableVisProvider(Private) {
+  //const VisFactory = Private(VisFactoryProvider);
   const VisType = Private(VisVisTypeProvider);
   const TemplateVisType = Private(TemplateVisTypeProvider);
   const Schemas = Private(VisSchemasProvider);
 
+  // define the EnhancedTableVisProvider which is used in the template
+  // by angular's ng-controller directive
+
+  // return the visType object, which kibana will use to display and configure new
+  // Vis object of this type.
   return new TemplateVisType({
+    type: 'table',
     name: 'enhanced-table',
     title: 'Enhanced Table',
     image,
     description: 'Same functionality than Data Table, but with enhanced features like computed columns and filter bar.',
-    category: VisType.CATEGORY.DATA,
-    template: visTemplate,
+    category: VisType.CATEGORY.MAP,
```

Comment:
Why did you change the visualization category ?
Could you restore "DATA" category ?

Reply:
Done.